### DISCUSSION TITLE
[Solved] Ugreen 2,5 Gbe works only with 1 Gbe

### DISCUSSION THREAD

**Discussion by Schockwelle** [bb-qq/r8152#355] [Q&A] [answered]
Hello everyone,
I bought two Ugreen 2.5 Gbe adapters (2023 model) and a 2.5 Gbe switch and installed everything. The switch, my router and the adapter to my Mac work with 2.5 Gbe.

The bb-qq driver on my new DS224+ is installed and the Ugreen 2.5 Gbe adapter is connected. As you can see in the photos, it is also recognized.

However, as you can see in the picture, it only works with 1 Gbe. Does anyone have a tip for me on how I can switch it to 2.5 Gbe? I have already replaced the network cable between the DS224+ and the switch several times with other cables (Cat 5e / Cat 6), but unfortunately without success.

Regards

<img width="915" alt="Monosnap nail — ssh admin@192 168 1 11 — 191×59 2023-11-28 12-00-28" src="https://github.com/bb-qq/r8152/assets/4801758/c8a2da5c-c42e-4328-9992-2bf1571fe1a9">

<img width="956" alt="Monosnap Diskstation - Synology NAS 2023-11-28 11-48-26" src="https://github.com/bb-qq/r8152/assets/4801758/3ba73fe1-fdb4-4a49-ba1c-3ec8f6ef135e">

**Comment by Sjekke** [comment]
Canyou force your switch to use 2,5gbit?

**Reply by Schockwelle** [reply]
Unfortunately not, it is an unmanaged switch (D-Link DMS-108 8-Port Multi-Gigabit Unmanaged Desktop Switch).
Only blue led (2,5gbe) is on for my Mac and router. NAS led only lights up green for 1gbe.

**Reply by elnono** [reply]
which model of Ugreen 2.5 Gbe adapters (2023 model) do you use?
it's a usb-c adapters?

**Reply by TheS1R** [reply]
I am successfully using **UGREEN Model CM648** USB-C to 2.5 Gbps Ethernet adapter with RTL8156BG (_lower power, less heat_) chipset ([https://www.amazon.com/gp/product/B0CD1FDKT1](url)) with XAOSUN One-Sided SuperSpeed Data Sync [10Gbps] USB-C Female to USB-A Male Adapter ([https://www.amazon.com/dp/B085ZYJWDW](url)) with my DS718+, achieving a full 2.5 Gbps connection. It is connected to network via TP-Link TL-SG108-M2 8-port 2.5 Gbps unmanaged switch ([https://www.amazon.com/dp/B08ZHZL5Q7](url)) — I have found that TP-Link unmanaged switches are one of the few that transparently support the forwarding of VLAN tags to/from devices..

**Reply by Schockwelle** [reply]
Yes, it is USB-C with with a little Adapter from USB-C to USB-A.

**Reply by TheS1R** [reply]
Try reversing the direction of the USB-C to USB-A adapter — some of them are one-way for high-speed data.

**Reply by elnono** [reply]
lusb on your screenshot return 480mbits/s for the adapter, so it's recognized as usb 2.0

try 'cat /sys/bus/usb/devices/1-2/version' to be sure.

In usb-c adapter there is a little chip to detect the connection specs with the pair.
There is maybe an issue between your nas and the usb -c to usb-A dongle.

**Reply by Schockwelle** [reply]
@TheS1R & @elnono 
Yesterday I ordered a new adapter.
I will report whether it worked.

@elnono 
"cat /sys/bus/usb/devices/1-2/version" returns "2.10".

**Reply by Schockwelle** [reply]
Now it works with a new  10g adapter:

 3.20 5000MBit/s 256mA 1IF  (Realtek USB 10/100/1G/2.5G LAN 4013000001)
 
Thank you.

<img width="885" alt="Monosnap Diskstation - Synology NAS 2023-11-30 18-19-44" src="https://github.com/bb-qq/r8152/assets/4801758/ad7a3132-1ff8-4809-916f-48d218b198c3">

**Reply by ParPeter** [reply]
I'm also having a problem with the Ugreen adapter, except that in my case they are recognised as providing 2.5Gb connections - I have two of them.  
But a transfer between two NAS (DS-923+ and DS-920+) is running at only 10-12Mbs according to the performance monitor. (I've not been able to find out or work out how to run iperf/iperf3.) There's nothing else of consequence running over the ethernet.
![Screenshot 2023-12-25 104109](https://github.com/bb-qq/r8152/assets/153178127/77f9ce44-6d6d-4b09-b3a9-119623d4f0c3)
![Screenshot 2023-12-25 112043](https://github.com/bb-qq/r8152/assets/153178127/b19a5b50-023b-42ef-8ccf-1fa9c403c05e)
![Screenshot 2023-12-25 113234](https://github.com/bb-qq/r8152/assets/153178127/732e7cc3-88a8-4bce-8b6c-63f9cbcaa8bc)

**Reply by ParPeter** [reply, marked as answer]
Think I've sorted this now, when I found that the cable from the DS-923+ was connected to the switch at 1Gb. Disconnected and reconnected and it reconnected at 2.5Gb. Upload speed increased to approx 150MB/s within a few minutes.
(I've yet to find a 2.5Gb switch that allows connections to be made at the back while displaying status at the front.)

**Reply by ParPeter** [reply]
Found out how to run iperf3 and can advise that the Ugreen device runs at just under 2.5Gb.

**Reply by Doughboy68** [reply]
I think the original issue is that in some cases the Ugreen will not be seen as a USB 3 device eg. 5000Mbps, and when it is USB 2 480Mbps it only does 1Gbps on its interface instead of 2.5Gbps. I am going to try a few things and will follow up if work.

**Reply by tabulaRasaGit** [reply]
> Try reversing the direction of the USB-C to USB-A adapter — some of them are one-way for high-speed data.

I've had the same problem with the Ugreen Usb-C adapter (only recoginzed as USB2.0 device). I also used an Usb-C to Usb-A adapter (from Ugreen - specified for 10gbps). -> the solution was reversing the Usb-C connector. Thanks a lot!

**Comment by Doughboy68** [comment]
> Hello everyone, I bought two Ugreen 2.5 Gbe adapters (2023 model) and a 2.5 Gbe switch and installed everything. The switch, my router and the adapter to my Mac work with 2.5 Gbe.
> 
> The bb-qq driver on my new DS224+ is installed and the Ugreen 2.5 Gbe adapter is connected. As you can see in the photos, it is also recognized.
> 
> However, as you can see in the picture, it only works with 1 Gbe. Does anyone have a tip for me on how I can switch it to 2.5 Gbe? I have already replaced the network cable between the DS224+ and the switch several times with other cables (Cat 5e / Cat 6), but unfortunately without success.
> 
> Regards
> 
> <img alt="Monosnap nail — ssh admin@192 168 1 11 — 191×59 2023-11-28 12-00-28" width="915" src="https://private-user-images.githubusercontent.com/4801758/286227480-c8a2da5c-c42e-4328-9992-2bf1571fe1a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUxMzc2MDUsIm5iZiI6MTcwNTEzNzMwNSwicGF0aCI6Ii80ODAxNzU4LzI4NjIyNzQ4MC1jOGEyZGE1Yy1jNDJlLTQzMjgtOTk5Mi0yYmYxNTcxZmUxYTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMTNUMDkxNTA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGViYjMzNWZhODBhMmI3M2RmYzM1NjEyNjFjOTUwNzFkNzEwMTY2YWQ1ZDFiZmJkZjZiMDc4NWU3ZmExMDg3MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.eNX9EYOVSp8jahs2wrfQHj1uSv4m3lyTf5RZQmYwfao"> <img alt="Monosnap Diskstation - Synology NAS 2023-11-28 11-48-26" width="956" src="https://private-user-images.githubusercontent.com/4801758/286227485-3ba73fe1-fdb4-4a49-ba1c-3ec8f6ef135e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUxMzc2MDUsIm5iZiI6MTcwNTEzNzMwNSwicGF0aCI6Ii80ODAxNzU4LzI4NjIyNzQ4NS0zYmE3M2ZlMS1mZGI0LTRhNDktYmExYy0zZWM4ZjZlZjEzNWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDExMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMTNUMDkxNTA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTI0MDM2OTk5ZjE5MDUxMDVhMWUyNjNkMjk3Mjk0YTQ0NzQ5NzBhYjM2ZWZhMzQ3ZTQwZTZhYjVjMWM5MmE3YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.2CqLE0kZWWfuiBMPozJkE1u2aohKeq06G2fOF3Nw6EI">

I have a 1520+ and can only get it to negotiate at switch 1Gbps
I look at  your lsusb and notice the USB port is only 480Mbps, and so is mine. How do we force it to connect at USB 3 speeds? My actual transfer is around 40MB/s when using that interface so it definitely is only USB2. Maybe is issue with USBC to USBA adapter I am using? I have a 1Gbps USB type A ethernet adapter and when I plugged it into the Synology it shows 5000Mbps with lsusb... therefore I think these Ugreen with USBC and an A adapter may not work with Synology... however the UGreen works fine on a PC.

**Comment by skipdashu** [comment]
This one reports as and shows at switch with 2.5Gbps light on but actual through put is under 1Gbps on my DS224+.  Like 260KB/s.  I get much better xfer rate using it's stock 1Gbps ports.   This was on rear USB port.  Retesting now using front port which lsusb reports as 5MB.

Front USB port a wee bit better but mostly 600K to 900K best with rsync xfer.  I'll see if can do any better plugged into my Linux desktop box.

https://www.amazon.com/dp/B0BNHK5RGL?ref=ppx_yo2ov_dt_b_product_details&th=1

Ordered this one to give it a try:
https://www.amazon.com/gp/product/B09V2J992N/ref=ppx_yo_dt_b_asin_title_o00_s00?ie=UTF8&psc=1
